### DISCUSSION TITLE
Playlist changes

### DISCUSSION THREAD

**Discussion by TheSimTek** [music-assistant/support#2981] [Q&A] [answered]
Hello Developers,

I have started using Music Assistant to play music from my NAS. It is amazing but I have a few questions.

One - my playlist is 500 songs so far but I am unable to delete one or two tracks that I have added twice. It either does nothing or comes back with an error. Will try to capture this soon.

Two - my playlist has evolved from its original title and I wish to rename it. How can this be done?

Three - where is Music Assistant located on the system? Can the playlist file be edited manually?

Thanks for your support and amazing add on.  Hope these ideas can be implemented in future versions.

**Comment by OzGav** [comment, marked as answer]
I will look into #1 soon although there is this issue https://github.com/music-assistant/hass-music-assistant/issues/2903

#2 you can’t rename per se but you can select all the contents of the playlist (CTRL-A) and then create a new playlist containing those tracks then delete the old one.

#3 I assume you mean when running as an add on. It is in a docker container on the HA host. We don’t supply support for manually manipulating that as it will likely result in more workload when people mess it up.

**Reply by TheSimTek** [reply]
Thank you OzGav.

It appears issue #1 is very similar to that you linked in #2903.  Now over 500 songs the ones I have tried to delete do not get removed.  I no longer get the "List assignment index out of range" error that I tried to capture, just nothing happens.

For #2 I may create a new playlist as it evolves.

#3 was in case I could edit the data file to delete those I could not in #1. 

Thanks for looking into the first issue.

Started using this as an add on for HA a few weeks ago and working on automations to play my music. 

Excellent work.  Thank you.

**Reply by OzGav** [reply]
I don't have a problem removing the contents of a playlist except as per the issue the track after the one selected is removed. I am interested where you say you added a track twice. Do you mean a similar track (remaster or something like that) as you should not be able to add two identical tracks, that should result in an info entry in the logs?
![image](https://github.com/user-attachments/assets/c41c6a8a-0d74-4a67-967f-b50f7aaefb7b)

**Reply by TheSimTek** [reply]
If I have two songs of the same name but from the original album and then off a best of album I have tried to remove one of them. 

it does nothing.  After I refresh or come back to Music Assistant they are both still there. 

When I can get the logs from the system I will post them for you. 

Thanks.

**Reply by OzGav** [reply]
When and how did you create this playlist?

**Comment by TheSimTek** [comment]
I created the playlist with Music Assistant within Home Assistant.  Going through my albums, I click and add to playlist.  I started two weeks ago with server version 2.2.6 and since upgraded to 2.2.7 where it hasn't changed the issue I am having. 

Refer to screen shot for a double up of the same track on two different albums.  I have tried to remove one over several attempts but both remain in the playlist.

![playlist](https://github.com/user-attachments/assets/436765d1-983c-4d42-83c7-ec8f3cfc0959)

I added some songs this morning (refer to logs) and then attempted to delete the second track "Six Months in a Leaking Boat" (236) where nothing happened.  There is no report in the log below of the deletion and after a refresh it is still there.

![playlog](https://github.com/user-attachments/assets/b24ed6f1-7ef1-4f50-b57f-899199ad4b68)

As stated, it recognises the same track "Better" and does not add it.

**Reply by OzGav** [reply]
Have you tagged the "Six Months in a Leaky Boat" tracks with Picard? if so, can you check the tags to confirm if they are identically tagged (except for the album details)?

**Comment by TheSimTek** [comment]
This issue is not the tagging convention of the songs.  Regardless of what they are I should be able to delete one of them.  Here is the info:

![music](https://github.com/user-attachments/assets/02555d0b-6082-45cd-be35-1d2bc077e1e0)

I have selected a few other tracks I no longer want in the playlist.  I cannot delete them either.  They only appear once but nothing happens when "remove from playlist" is selected other than a blue working circle occasionally in the top right corner then nothing.  The tracks remain in the playlist.

**Reply by OzGav** [reply]
Have you pressed the refresh icon at the top of the view?

if those are the only tags on the track then it’s possible MA sees them as different (to avoid problems you need to fully tag them with the MB identifiers). So I can forsee a bug where they were allowed to be added to the playlist but then the duplicate name (with no other tags) is causing problems.

**Reply by TheSimTek** [reply]
Thanks OzGav,
I have hit the refresh icon. No change.

I have also tried my desktop PC, laptop and the phone with the same results incase of cached data. 

I have added a few tracks, then tried to delete one without success.  Log shows addition but no removal.  Same as above log image

**Reply by OzGav** [reply]
What version of the server are you using. Sorry I see you are on stable. Can you try the beta

**Comment by TheSimTek** [comment]
Started playlist on 2.2.6 and updated over weekend to 2.2.7.

**Reply by OzGav** [reply]
Yes please try the beta. Also try fully tagging three tracks. Create a new playlist and add those and then try and delete one (noting the existing issue don’t try and delete the last one and expect the track after the one you select to be deleted)

**Reply by TheSimTek** [reply]
Will attempt during the week and let you know what happens

**Reply by TheSimTek** [reply]
Uninstalled server and installed beta version 2.3.0b29.  Here I managed to create a 4 song playlist and remove one track.  As stated, it ends up being the one below selected.  Log files show the 4 entries but does not show the removal of a song from the playlist.

I have restored by backup and tested again.  It appears if I search for a file name and identify two in the playlist I am unable to delete them.  However, if I know the number I want to remove is 236 then I scroll down to it and delete 235.  This worked.  The double entry is no longer present.

Could be a glitch in removal during a search function.

**Reply by OzGav** [reply]
OK what if you did the search but tried to delete the first in the returned results? Is that works then this is the bug mentioned.

**Comment by TheSimTek** [comment]
Try to clear it up.  

If I have two songs the same but from different albums and the search by name shows both of them then I cannot delete either one.

However, if I know the number in the list and scroll through until I find that number (or the one above it) I can successfully delete the track.

**Reply by OzGav** [reply]
OK moving this to an issue
